src/BuildScriptGenerator/Python/PythonPlatformDetector.cs

```diff
@@ -29,46 +29,52 @@ internal class PythonPlatformDetector : IPlatformDetector
         public PlatformDetectorResult Detect(RepositoryContext context)
         {
             var sourceRepo = context.SourceRepo;
-            if (!sourceRepo.FileExists(PythonConstants.RequirementsFileName)
-                && !sourceRepo.FileExists(PythonConstants.SetupDotPyFileName))
+
+            var isPythonApp = IsPythonApp(sourceRepo);
+
+            // This detects if a runtime.txt file exists and if that is a python file
+            var versionFromRuntimeFile = DetectPythonVersionFromRuntimeFile(context.SourceRepo);
+            if (!isPythonApp && string.IsNullOrEmpty(versionFromRuntimeFile))
             {
-                _logger.LogDebug($"'{PythonConstants.SetupDotPyFileName}' or '{PythonConstants.RequirementsFileName}' " +
-                    $"does not exist in source repo");
                 return null;
             }
-            else if (!sourceRepo.FileExists(PythonConstants.RequirementsFileName)
-                && sourceRepo.FileExists(PythonConstants.SetupDotPyFileName))
+
+            return new PlatformDetectorResult
             {
-                _logger.LogInformation($"'{PythonConstants.RequirementsFileName} doesn't exist in source repo.' " +
-                    $"Oryx will try to build from '{PythonConstants.SetupDotPyFileName}'that exists in source repo");
+                Platform = PythonConstants.PlatformName,
+                PlatformVersion = versionFromRuntimeFile,
+            };
+        }
+
+        private bool IsPythonApp(ISourceRepo sourceRepo)
+        {
+            if (sourceRepo.FileExists(PythonConstants.RequirementsFileName))
+            {
```

Comment:
if we have an app where req.txt doesnt exist but setup.py exists .. we  are not detecting it as python app